src/test/java/PanbyteTest.java

```diff
@@ -0,0 +1,439 @@
+        String expectedOutput = "test" + System.lineSeparator() + "macka" + System.lineSeparator() + "pes";
+        String actualOutput = getOutputOfProgramCall(args);
+        assertEquals(expectedOutput, actualOutput);
+    }
+
+    @Test
+    void testFromFileWithSingleCharToStdout() {
+        URL url = getClass().getResource("input/inputFileWithSingleCharDelimiter.txt");
+
+        String[] args = {"-i", url.getPath(), "-f", "bits", "--delimiter=q", "-t", "bytes" };
+        String expectedOutput = "OKqPESqMACKAqoK";
+        String actualOutput = getOutputOfProgramCall(args);
+        assertEquals(expectedOutput, actualOutput);
+    }
+
+    @Test
+    void testFromFileWithMultipleCharsToStdout() {
+        URL url = getClass().getResource("input/inputFileWithMultipleCharsDelimiter.txt");
+
+        String[] args = {"-d", "I _ I", "-i", url.getPath(), "-f", "hex", "-t", "hex" };
+        String expectedOutput = "74657374I _ I706573I _ I6d61636b61";
+        String actualOutput = getOutputOfProgramCall(args);
+        assertEquals(expectedOutput, actualOutput);
+    }
+
+    @Test
+    void testFromStdoutToFileWithSingleChar() {
+
+        String outputFolderPath = getClass().getResource("output").getPath();
+        String fileName = "outputFileWithSingleCharDelimiter.txt";
+
+        String[] args = {"-o", outputFolderPath + File.pathSeparator
+                + fileName, "-f", "hex", "-t", "bytes"};
+        String expectedOutput = "test";
+
+        String actualOutput;
+
+        String echo = "74657374";
+        try (ByteArrayInputStream byteArrayInputStream =
+                     new ByteArrayInputStream(echo.getBytes(StandardCharsets.UTF_8))) {
+            System.setIn(byteArrayInputStream);
+            PanByte.main(args);
+        } catch (IOException e) {
+            e.printStackTrace();
+            fail("File exception");
+        }
+
+        try (BufferedReader reader = new BufferedReader(
+                new FileReader(outputFolderPath + File.pathSeparator + fileName)
+        )) {
+            actualOutput = reader.readLine();
+            assertEquals(expectedOutput, actualOutput);
+        } catch (IOException e) {
+            e.printStackTrace();
+            fail("File exception");
+        }
+        File outputFile = new File(outputFolderPath + File.pathSeparator + fileName);
+        outputFile.delete();
+    }
+
+
+    //endregion
+    private String getOutputOfProgramCall(String echo, String[] args) {
+        try (
+                ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(echo.getBytes(StandardCharsets.UTF_8));
+                ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+                PrintStream printStream = new PrintStream(byteArrayOutputStream, true, StandardCharsets.UTF_8)
+        ) {
+            System.setIn(byteArrayInputStream);
+            System.setOut(printStream);
+            PanByte.main(args);
```

Comment:
**suggestion:** you are not checking the return value of main.

I would let the operating system spawn the new process instead of just calling main. These are not true integration tests.